yt/data_objects/data_containers.py

```diff
@@ -782,16 +782,7 @@ def create_firefly_object(
             >>> reader.dumpToJSON()
         """
 
-        ## attempt to import Firefly
-        try:
-            from Firefly.data_reader import ParticleGroup, Reader
-        except ImportError as e:
-            raise ImportError(
-                "Can't find Firefly, ensure it "
-                "is in your python path or install it with "
-                "`python -m pip install Firefly-vis`. It is also available "
-                "on github at github.com/ageller/Firefly"
-            ) from e
+        from yt.utilities.on_demand_imports import _firefly as firefly
```

Comment:
Thanks for patching this. Another remark is that the way we implement our "on_demand_imports" module, this import is perfectly safe at the top level of the module, it doesn't need to be embedded locally. Can you move it please ? (we're getting there :))

Reply:
good call